arch/arc/core/fast_irq.S

```diff
@@ -178,7 +178,6 @@ _firq_no_reschedule:
 #if CONFIG_RGF_NUM_BANKS == 1
 	_pop_irq_stack_frame
 #endif
-	lr ilink, [_ARC_V2_ERET]
```

Comment:
Do you know why ERET was used/copied before? Was it used as scratch register earlier, but now no longer? Or, could there be some other reason why ilink doesn't contain the return address?

Reply:
In EM's ISA,  ERET has a copy of ilink, but in HS's ISA,  ERET does not have. According to @abrodkin's suggestion , I do some changes in previous PR, and do not rely on that ERET has a copy of ilink. 

The removed codes are some codes should be removed before.